LiskHQ/lisk-hub
# Fix Ledger detection - Closes #1976

### What issue have I solved?

#1976 

### How have I implemented/fixed it?

Updated how the Ledger connection was being detected, to only check if it is connected, instead of checking if the Lisk App is open, and added a function to be able to update the device on the open App screen, passing an event through ipc.  
And handling going back on the multistep component if the devices are disconnected.

**Known Issue**: Detecting that the app was opened is buggy, because of how the events of Ledger are recognize by the HID, every time the app is open or closed, it sends a remove and add event to the USB

### How has this been tested?

1. Run this build on electron
1. Go to `/hw-wallet-login-v2`
1. Connect and unlock Ledger
1. Should go to the page asking for opening the app on ledger.
1. Opening the app should go to select account page.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
